baseplate/observers/metrics.py

```diff
@@ -29,82 +31,105 @@ class MetricsBaseplateObserver(BaseplateObserver):
 
     """
 
-    def __init__(self, client: metrics.Client):
+    def __init__(self, client: metrics.Client, sample_rate: float = 1.0):
         self.client = client
+        self.sample_rate = sample_rate
+
+    @classmethod
+    def from_config_and_client(
+        cls, raw_config: Optional[config.RawConfig], client: metrics.Client
+    ) -> "MetricsBaseplateObserver":
+        sample_rate = 1.0
+        if raw_config:
+            cfg = config.parse_config(
+                raw_config,
+                {"metrics_observer": {"sample_rate": config.Optional(config.Float, default=1.0)}},
```

Comment:
it'd be clearer for this to be `config.Percent` so the config looks like `metrics_observer.sample_rate = 37%` which is more self-describing than wondering what the range of the value is. the output will still be exactly a float between 0 and 1.0 like you've got here.

Reply:
looks like this still needs to be done?